pkg/config/config_template.yaml

```diff
@@ -127,6 +127,18 @@ api_key:
 # number of check runners instances concurrently. Setting the value to 1
 # would result in checks running sequentially.
 #
+# NOTE: due to the nature of the python interpreter and it's global interpreter
+# lock (GIL) only one native thread may be running on the python interpreter at
+# a time. The introduction of concurrent check runners implies concurrently
+# executing checks must contend for the GIL. This has the side-effect of increasing
+# wall-time execution times for the checks. The aggregate system/user-time for all
+# checks however, should be decreased. Concurrency works particularly well with I/O
+# bound tasks (or at least IO intensive) as the GIL will typically be released while 
+# waiting for IO, allowing for other checks to grab a hold of the GIL and continue
+# execution. For CPU bound checks with a low IO activity, it becomes increasingly
+# likely for these check instances to hog the GIL delaying all other checks from
+# execution - essentially forcing a sequential behavior.
```

Comment:
it's actually a bit worse than just a sequential behavior in this case (GIL-related contention is CPU-costly). I think we can mention that CPU-intensive custom python checks that take more than 1 second to run may benefit from forcing a sequential behavior, i.e. setting `check_runners` to `1`.